libcrux-ml-dsa/src/simd/avx2/arithmetic.rs

```diff
@@ -72,3 +82,104 @@ pub fn shift_left_then_reduce<const SHIFT_BY: i32>(simd_unit: Vec256) -> Vec256
 
     mm256_sub_epi32(shifted, quotient_times_field_modulus)
 }
+
+// TODO: Revisit this function when doing the range analysis and testing
+// additional KATs.
+#[inline(always)]
+pub fn infinity_norm_exceeds(simd_unit: Vec256, bound: i32) -> bool {
+    let absolute_values = mm256_abs_epi32(simd_unit);
+
+    // We will test if |simd_unit| > bound - 1, because if this is the case then
+    // it follows that |simd_unit| >= bound
+    let bound = mm256_set1_epi32(bound - 1);
```

Comment:
Ah, I'm not sure if it can underflow, but I put the TODO there since I haven't tried to properly calculate the representative ranges that this function will see.